build_tools/upload_logs_to_s3.py

```diff
@@ -0,0 +1,90 @@
+# === Updated build_tools/upload_logs_to_s3.py ===
+#!/usr/bin/env python3
+"""
+upload_logs_to_s3.py
+
+Uploads log files and index.html to an S3 bucket using the AWS CLI.
+"""
+
+import os
+import sys
+import glob
+import shutil
+import argparse
+import subprocess
+from pathlib import Path
+
+
+def log(*args, **kwargs):
+    print(*args, **kwargs)
+    sys.stdout.flush()
+
+
+def check_aws_cli_available():
+    if shutil.which("aws") is None:
+        log("[ERROR] AWS CLI not found in PATH.")
+        sys.exit(1)
+
+
+def run_aws_cp(src: str, dest: str, content_type: str = None):
+    cmd = (
+        ["aws", "s3", "cp", src, dest, "--recursive"]
+        if os.path.isdir(src)
+        else ["aws", "s3", "cp", src, dest]
+    )
+    if content_type:
+        cmd += ["--content-type", content_type]
+    try:
+        log(f"[INFO] Running: {' '.join(cmd)}")
+        subprocess.run(cmd, check=True)
+    except subprocess.CalledProcessError as e:
+        log(f"[ERROR] Failed to upload {src} to {dest}: {e}")
+
+
+def upload_logs_to_s3(s3_base_path: str, build_dir: Path):
+    log_dir = build_dir / "logs"
+
+    if not log_dir.is_dir():
+        log(f"[INFO] Log directory {log_dir} not found. Skipping upload.")
+        return
+
+    # Upload .log files
+    log_files = list(log_dir.glob("*.log"))
+    if not log_files:
+        log("[WARN] No .log files found. Skipping log upload.")
+    else:
+        run_aws_cp(str(log_dir), s3_base_path, content_type="text/plain")
+
+    # Upload index.html
+    index_path = log_dir / "index.html"
+    if index_path.is_file():
+        index_s3_dest = f"{s3_base_path}/index.html"
+        run_aws_cp(str(index_path), index_s3_dest, content_type="text/html")
+        log(f"[INFO] Uploaded {index_path} to {index_s3_dest}")
+    else:
+        log("[INFO] No index.html found. Skipping index upload.")
+
+
+def main():
+    check_aws_cli_available()
+
+    parser = argparse.ArgumentParser(description="Upload logs to S3.")
+    parser.add_argument(
+        "--build-dir",
+        type=Path,
+        default=Path("build"),
```

Comment:
This should anchor itself on the repository root, so running

```bash
cd build_tools
python upload_logs_to_s3.py
```

Doesn't search for logs in `build_tools/build`.

Reply:
Updated